### ISSUE TITLE
Can't change the Inbox directory for Calibre plugin when I change the SD Card

### ISSUE THREAD

**Issue by anagonzalez133** [koreader/koreader#6943]
* KOReader version: 2020.11
* Device: Motorola G9 Play with Android 10

#### Issue
I've installed the app and set it up with a SD Card. I've set the home directory and the Calibre plugin inbox directory to a directory in the SD Card. Then I have received a new SD Card, switched the cards in the phone, and now I can't change the Inbox directory in the Calibre plugin, it continues going to the old path in the old SD Card and there's no way to navigate to the internal storage or the new SD card. I have even tried going back to the old card, but it keeps saying that the path is not accesible and I can't change it.

#### Steps to reproduce
Set Calibre plugin Inbox directory to a path in the SD Card. Afterwards change the SD Card to another one.

**Comment by pazos** [comment]
You shouldn't set calibre inbox in the sdcard. These are usually read only.

About the rest of the issue: >  it continues going to the old path in the old SD Card and there's no way to navigate to the internal storage or the new SD card

Could you attach screenshots?

**Comment by anagonzalez133** [comment]
Thanks pazos. I'm trying to set the Android directory of Koreader as the input directory. I've created the directory using Koreader file browser, and set it up as home directory. Is when I try to set it up as Input directory for the Calibre plugin when I have problems, even after uninstalling and reinstalling the app again. I'm going to upload some screenshots

**Comment by anagonzalez133** [comment]
This is the preselected directory as Input directory. I can't navigate to another directory, I can only go to root / 
![Screenshot_20201204-173633-191](https://user-images.githubusercontent.com/10425604/101189725-b2844680-3657-11eb-8fb0-6e77fb79992e.png)

**Comment by anagonzalez133** [comment]
This is what happens when I try to navigate to another directory trying to change the input directory, I can only go to root and offered only to select that directory. 
![Screenshot_20201204-174329-675](https://user-images.githubusercontent.com/10425604/101190690-03e10580-3659-11eb-89ad-da558f66a666.png)

**Comment by anagonzalez133** [comment]
This is the new directory on the new SDCard. I created it using Koreader file browser, and I had no problem setting it up as Home directory 
![Screenshot_20201204-175140-454](https://user-images.githubusercontent.com/10425604/101191401-ee201000-3659-11eb-8237-39d16d124160.png)

**Comment by pazos** [comment]
Hey @anagonzalez133, thanks for the screenshots. To fix your issue use any text editor and edit the file `/sdcard/koreader/settings.reader.lua`

Inside there's a line called something like
```lua
["inbox_dir"] = "/storage/UUID/path",
```

Change that to
```lua
["inbox_dir"] = "/storage/emulated/0",
```

exit the program and start again

What I still don't know is how you did to set the inbox dir in the sdcard the first time. Maybe I'm missing something. If the previous change fixes your issue please double check that you're not able to setup the inbox_dir to sdcard again.

And finally, double check that you're not able to set the inbox dir in the sdcard even if you remove the entire line from `/sdcard/koreader/settings.reader.lua`

**Comment by pazos** [comment]
BTW: you shouldn't create a library under Android/data/org.koreader.launcher. Files there will be removed if/when the program is deleted. Also they're not guaranteed to stay safe even if the program is installed, because android can reclaim space there if your device has low storage available.

If you update to one of the new nightlies you can change from internal storage and sdcard in the filemanager: see https://github.com/koreader/koreader/pull/6919#issue-528770164. Even then you won't be able to create directories outside Android/data/org.koreader.launcher on most phones. So the final solution is the same:

Keep your wireless library in the internal storage if possible. That's the only place where the program is granted to work as expected.

**Comment by anagonzalez133** [comment]
Thanks, changing the parameters in the file you told me I've been able to fix it.
Re using the SD card, thanks for the advice. I don't want to use the internal storage because I want to keep a bit large library in my phone, and the internal storage can become quite cramped with time. I'm trying to find a substitute for Calibre Companion and Bookari, which are the programs I was using in my old phone, but they are going to give me trouble or not work at all on Android 10. Koreader works partially for my use, but unfortunately it doesn't have access to metadata in Calibre I find useful (mainly a custom column for Read/Unread classification, and the date the book was added to Calibre, something useful when you keep a large library and are trying to decide what next Unread book you want to read). If using the SD card gives me problems, I'll have to continue looking the ideal alternative for my use. There are some alternatives for Bookari, but it's more difficult finding something instead of CC.
